src/main/java/ch/njol/skript/lang/SyntaxElementInfo.java

```diff
@@ -69,4 +78,42 @@ public String[] getPatterns() {
 	public String getOriginClassPath() {
 		return originClassPath;
 	}
+	
+	@SuppressWarnings("unchecked")
+	@Contract("_ -> new")
+	public static <I extends SyntaxElementInfo<E>, E extends SyntaxElement> I fromModern(SyntaxInfo<? extends E> info) {
+		if (info instanceof BukkitSyntaxInfos.Event) {
+			BukkitSyntaxInfos.Event<?> event = (BukkitSyntaxInfos.Event<?>) info;
+
+			// We must first go back to the raw input
+			String rawName = event.name().startsWith("On ")
+					? event.name().substring(3)
+					: "*" + event.name();
+			SkriptEventInfo<?> eventInfo = new SkriptEventInfo<>(rawName, event.patterns().toArray(new String[0]), event.type(),
+				event.origin().name(), (Class<? extends Event>[]) event.events().toArray(new Class<?>[0]))
+				.since(event.since())
+				.documentationID(event.documentationId())
+				.description(event.description().toArray(new String[0]))
+				.examples(event.examples().toArray(new String[0]))
+				.keywords(event.keywords().toArray(new String[0]))
+				.requiredPlugins(event.requiredPlugins().toArray(new String[0]));
```

Comment:
```suggestion
					event.origin().name(), (Class<? extends Event>[]) event.events().toArray(new Class<?>[0]))
					.since(event.since())
					.documentationID(event.documentationId())
					.description(event.description().toArray(new String[0]))
					.examples(event.examples().toArray(new String[0]))
					.keywords(event.keywords().toArray(new String[0]))
					.requiredPlugins(event.requiredPlugins().toArray(new String[0]));
```